File: carlos.cirello/stylelang/ast/defaultnode.go
package ast

type DefaultNode struct {
	questionType string
	widget       string
}

func NewDefaultNode(questionType, widget string) *DefaultNode {
	return &DefaultNode{questionType, widget}
}

func (d *DefaultNode) QuestionType() string {
	return d.questionType
}

func (d *DefaultNode) Widget() string {
	return d.widget
}

File: kennedy-langlotz/KLQ/src/com/kls/ast/node/DefaultNode.java
package com.kls.ast.node;

import com.klq.logic.question.Type;
import com.kls.ast.IVisitor;
import com.common.Location;

import java.util.List;

/**
 * Created by Timon on 03.03.2015.
 */
public class DefaultNode extends ASelectorNodeBase {
    private final Type type;

    public DefaultNode(Type type, List<DeclarationNode> declarations, Location location) {
        super(declarations, location);
        this.type = type;
    }

    @Override
    public <T> T accept(IVisitor<T> visitor) {
        return visitor.visit(this);
    }

    public Type getType() {
        return type;
    }
}
